src/Orchard.Web/Modules/Orchard.Localization/Services/LocalizationService.cs

```diff
@@ -87,5 +91,51 @@ public class LocalizationService : ILocalizationService {
             // Warning: May contain more than one localization of the same culture.
             return query.List().ToList();
         }
+
+        bool ILocalizationService.TryGetRouteForUrl(string url, out AutoroutePart route) {
+            route = _contentManager.Query<AutoroutePart, AutoroutePartRecord>()
+                .ForVersion(VersionOptions.Published)
+                .Where(r => r.DisplayAlias == url)
+                .List()
+                .FirstOrDefault();
+
+            if (route == null)
+                route = _homeAliasService.GetHomePage(VersionOptions.Latest).As<AutoroutePart>();
+            return route != null;
+        }
+
+        bool ILocalizationService.TryFindLocalizedRoute(ContentItem routableContent, string cultureName, out AutoroutePart localizedRoute) {
+            if (!routableContent.Parts.Any(p => p.Is<ILocalizableAspect>())) {
+                localizedRoute = null;
+                return false;
+            }
+
+            IEnumerable<LocalizationPart> localizations = ((ILocalizationService)this).GetLocalizations(routableContent, VersionOptions.Published);
+
+            ILocalizableAspect localizationPart = null, siteCultureLocalizationPart = null;
+            foreach (LocalizationPart l in localizations) {
+                if (l.Culture.Culture == cultureName) {
```

Comment:
We are currently normalizing these in OC, should it compare with ignore case?

Reply:
Good idea! Updated in a5e4f1c147e732371a3966df8551e77c08260bbf.